frontend/src/components/forms/CreateResident.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useEffect, useState } from "react";
+import Select, { SingleValue } from "react-select";
+import {
+    Alert,
+    AlertDescription,
+    AlertIcon,
+    Box,
+    Button,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Input,
+    Modal,
+    ModalBody,
+    ModalContent,
+    ModalHeader,
+    ModalCloseButton,
+    ModalOverlay,
+    ModalFooter,
+    ScaleFade,
+    Divider,
+} from "@chakra-ui/react";
+
+import { AddIcon } from "@chakra-ui/icons";
+import { SingleDatepicker } from "chakra-dayzed-datepicker";
+import { Card, Col, Row } from "react-bootstrap";
+import CommonAPIClient from "../../APIClients/CommonAPIClient";
+
+import selectStyle from "../../theme/forms/selectStyles";
+import { singleDatePickerStyle } from "../../theme/forms/datePickerStyles";
+
+// TODO: Connect to Buidings table 
+const BUILDINGS = [
+    { label: "144", value: "144" },
+    { label: "362", value: "362" },
+    { label: "402", value: "402" },
+];
+
+const CreateResident = (): React.ReactElement => {
+
+    const [initials, setInitials] = useState("");
+    const [roomNumber, setRoomNumber] = useState("");
+    const [moveInDate, setMoveInDate] = useState(new Date());
+    const [building, setBuilding] = useState("");
+
+    const [initialsError, setInitialsError] = useState(false);
+    const [roomNumberError, setRoomNumberError] = useState(false);
+    const [moveInDateError, setMoveInDateError] = useState(false);
+    const [buildingError, setBuildingError] = useState(false);
+
+    const [isOpen, setIsOpen] = useState(false);
+    const [showAlert, setShowAlert] = useState(false);
+
+    const addResident = async () => {
+        await CommonAPIClient.createResident({
+            initial: initials.toLowerCase(),
+            roomNum: parseInt(roomNumber, 10),
+            dateJoined: moveInDate,
+            building,
+        });
+    };
+
+    const handleInitialsChange = (e: { target: { value: unknown } }) => {
+        const inputValue: string = e.target.value as string;
+        if (/^[a-z]{0,2}$/i.test(inputValue)) {
```

Comment:
another case to check for is if the Initial is one character, I was able to create a resident with one char:
```
 id | initial | room_num | date_joined | date_left | building 
----+---------+----------+-------------+-----------+----------
  1 | sc      |      123 | 2023-07-06  |           | 362
  2 | s       |      123 | 2023-07-06  |           | 144
  ```

Reply:
I will add another check for this.